test/e2e/install_test.go

```diff
@@ -24,90 +24,126 @@ const (
-				g.Expect(bd.Status.Conditions[1].Reason).To(Equal("InstallationSucceeded"))
-			}).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
+			By("eventually reporting a successful installation")
+			Eventually(checkOperatorInstalled(ctx, operator)).WithTimeout(defaultTimeout).WithPolling(defaultPoll).Should(Succeed())
 
+			By("verifying the expected bundle path")
+			bd := rukpakv1alpha1.BundleDeployment{}
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(operator), &bd)).To(Succeed())
+			Expect(bd.Spec.Template.Spec.Source.Type).To(Equal(rukpakv1alpha1.SourceTypeImage))
+			Expect(bd.Spec.Template.Spec.Source.Image.Ref).To(Equal("quay.io/operatorhubio/argocd-operator@sha256:1a9b3c8072f2d7f4d6528fa32905634d97b7b4c239ef9887e3fb821ff033fef6"))
+			Expect(operator.Status.ResolvedBundleResource).To(Equal(bd.Spec.Template.Spec.Source.Image.Ref))
+			Expect(operator.Status.InstalledBundleResource).To(Equal(bd.Spec.Template.Spec.Source.Image.Ref))
 		})
 		AfterEach(func() {
-			err := c.Delete(ctx, operatorCatalog)
-			Expect(err).ToNot(HaveOccurred())
-			err = c.Delete(ctx, operator)
-			Expect(err).ToNot(HaveOccurred())
+			Expect(c.Delete(ctx, operator)).To(Succeed())
+			Expect(c.Delete(ctx, operatorCatalog)).To(Succeed())
 		})
 	})
 })
+
+func checkCatalogUnpacked(ctx context.Context, catalog *catalogd.Catalog) func() error {
+	return func() error {
+		return checkCondition(ctx, catalog, catalogd.TypeUnpacked,
+			statusReason{metav1.ConditionTrue, catalogd.ReasonUnpackSuccessful},
+			statusReason{metav1.ConditionFalse, catalogd.ReasonUnpackFailed},
+		)
+	}
+}
+
+func checkOperatorResolved(ctx context.Context, op *operatorv1alpha1.Operator) func() error {
+	return func() error {
+		return checkCondition(ctx, op, operatorv1alpha1.TypeResolved,
+			statusReason{metav1.ConditionTrue, operatorv1alpha1.ReasonSuccess},
+			statusReason{metav1.ConditionFalse, operatorv1alpha1.ReasonResolutionFailed},
+		)
+	}
+}
+
+func checkOperatorInstalled(ctx context.Context, op *operatorv1alpha1.Operator) func() error {
+	return func() error {
+		return checkCondition(ctx, op, operatorv1alpha1.TypeInstalled,
+			statusReason{metav1.ConditionTrue, operatorv1alpha1.ReasonSuccess},
+			statusReason{metav1.ConditionFalse, operatorv1alpha1.ReasonInstallationFailed},
+		)
+	}
+}
+
+type statusReason struct {
+	status metav1.ConditionStatus
+	reason string
+}
+
+func checkCondition(ctx context.Context, obj client.Object, condType string, successStatusReason statusReason, failureStatusReasons ...statusReason) error {
+	if err := c.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
+		return err
+	}
+
+	var conditions []metav1.Condition
+	switch v := obj.(type) {
+	case *catalogd.Catalog:
+		conditions = v.Status.Conditions
+	case *operatorv1alpha1.Operator:
+		conditions = v.Status.Conditions
+	default:
+		return StopTrying(fmt.Sprintf("cannot get conditions for unknown object type %T", obj))
+	}
```

Comment:
An earlier iteration of this section of code called a separate function I had written that generically:
- converted to unstructured
- extracted "status.conditions" into `[]interface{}`
- marshaled that to JSON
- unmarshaled the JSON into []metav1.Condition

That code generically handled _anything_ that had `status.conditions` with type `[]metav1.Condition`, but it seemed a bit hacky/overkill. I'm happy to go back to that, or entertain other options as well.